src/e2e-test/features/bigquery/source/BigQueryToBigQuery_withConnections.feature

```diff
@@ -0,0 +1,101 @@
+    Then Navigate to the properties page of plugin: "BigQueryTable"
+    Then Verify toggle plugin property: "useConnection" is toggled to: "YES"
+    Then Verify plugin property: "connection" contains text: "bqConnectionName"
+    Then Verify input plugin property: "dataset" contains value: "dataset"
+    Then Verify input plugin property: "table" contains value: "bqSourceTable"
+    Then Validate output schema with expectedSchema "bqSourceSchema"
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "Wrangler"
+    Then Verify the Output Schema matches the Expected Schema: "bqSourceSchema"
+    Then Validate "Wrangler" plugin properties
+    Then Close the Plugin Properties page
+    Then Expand Plugin group in the LHS plugins list: "Sink"
+    Then Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Navigate to the properties page of plugin: "BigQuery2"
+    Then Click plugin property: "useConnection"
+    Then Click on the Browse Connections button
+    Then Select connection: "bqConnectionName"
+    Then Enter input plugin property: "referenceName" with value: "BQSinkReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqTargetTable"
+    Then Click plugin property: "truncateTable"
+    Then Click plugin property: "updateTableSchema"
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Connect plugins: "Wrangler" and "BigQuery2" to establish connection
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate records transferred to target table is equal to number of records from source table
+
+  @BQ_SOURCE_TEST @BQ_SINK_TEST
+  Scenario: To verify data is getting transferred from BigQuery to BigQuery with use connection functionality
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "BigQuery" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect plugins: "BigQuery" and "BigQuery2" to establish connection
+    Then Navigate to the properties page of plugin: "BigQuery"
+    Then Click plugin property: "switch-useConnection"
+    Then Click on the Browse Connections button
+    Then Select connection: "bqConnectionName"
```

Comment:
Is this scenario dependent on previous scenario for connection creation? If yes, this is bad.
We should create the connection in the beginning of scenario and delete at the end so that they are independent from each other.

Reply:
Updated to create connection as prerequisite in Before scenario hook.

Reply:
Can we add a step to delete the connection as well in after scenario hook?

Reply:
Common connection actions are required to delete the connection.
Raised e2e-framework PR https://github.com/cdapio/cdap-e2e-tests/pull/93

I will add after hook to delete the connection once above PR gets merged.

Reply:
Hook added to delete the connection after scenario.